### ISSUE TITLE
The LwFPlugin return a negative loss

### ISSUE THREAD

**Issue by ppppeng** [ContinualAI/avalanche#1018]
🐛 **Describe the bug**
I noticed the loss in tensorboard log is negative. And only ReplayPlugin and LwFPlugin were used. 

I debug the LwFPlugin code and find that the kl_div return a negative loss.

After the code "[ : . au ]" the sum of log_p or q is not 1, and may be this is a reason. 

🐝 **Expected behavior**
Does the negative loss is corret and make any sense? Or it should be fixed?

🦋 **Additional context**
The LwFPlugin version i used is shown below.

     def _distillation_loss(self, out, prev_out, active_units):
        """
        Compute distillation loss between output of the current model and
        and output of the previous (saved) model.
        """
        # we compute the loss only on the previously active units.
        au = list(active_units)
        log_p = torch.log_softmax(out / self.temperature, dim=1)[:, au]
        q = torch.softmax(prev_out / self.temperature, dim=1)[:, au]
        res = torch.nn.functional.kl_div(log_p, q, reduction="batchmean")        
        res=self.disstill_loss(log_p,q)
        return res

    def penalty(self, out, x, alpha, curr_model):
        """
        Compute weighted distillation loss.
        """
        if self.prev_model is None:
            return 0
        else:
            with torch.no_grad():
                if isinstance(self.prev_model, MultiTaskModule):
                    # output from previous output heads.
                    y_prev = avalanche_forward(self.prev_model, x, None)
                    # in a multitask scenario we need to compute the output
                    # from all the heads, so we need to call forward again.
                    # TODO: can we avoid this?
                    y_curr = avalanche_forward(curr_model, x, None)
                else:  # no task labels
                    y_prev = {"0": self.prev_model(x)}
                    y_curr = {"0": out}

            dist_loss = 0
            for task_id in y_prev.keys():
                # compute kd only for previous heads.
                if task_id in self.prev_classes:
                    yp = y_prev[task_id]
                    yc = y_curr[task_id]
                    au = self.prev_classes[task_id]
                    dist_loss += self._distillation_loss(yc, yp, au)
            return alpha * dist_loss

**Comment by ppppeng** [comment]
i updated the code 
    log_p = torch.log_softmax(out / self.temperature, dim=1)[:, au]
    q = torch.softmax(prev_out / self.temperature, dim=1)[:, au]
to
    log_p = torch.log_softmax(out[:, au] / self.temperature, dim=1)
    q = torch.softmax(prev_out[:, au] / self.temperature, dim=1)

and then run corretly.

**Comment by AntonioCarta** [comment]
Thanks for catching this. I will fix it.

**Comment by PabloMese** [comment]
That was helpful. Thanks @ppppeng
